consensus/vbft/service.go

```diff
@@ -1058,6 +1058,31 @@ func (self *Server) processProposalMsg(msg *blockProposalMsg) {
 		log.Errorf("BlockPrposalMessage check blocknum:%d,prevhash:%s,msg prevhash:%s", msg.GetBlockNum(), hex.EncodeToString(prevBlkHash[:4]), hex.EncodeToString(msgPrevBlkHash[:4]))
 		return
 	}
+	cfg := vconfig.ChainConfig{}
+	if blk.getNewChainConfig() != nil {
+		cfg = *blk.getNewChainConfig()
+	} else {
+		if blk.getLastConfigBlockNum() != math.MaxUint32 {
+			cfgBlock, _ := self.blockPool.getSealedBlock(blk.getLastConfigBlockNum())
+			if cfgBlock == nil {
+				log.Errorf("processProposalMsg failed to get cfg block: %d", blk.getLastConfigBlockNum())
+				return
+			}
+			if cfgBlock.Info.NewChainConfig != nil {
+				cfg = *cfgBlock.Info.NewChainConfig
+			} else {
+				log.Errorf("processProposalMsg failed to get chain config from config block:%d,%d", msgBlkNum, blk.getLastConfigBlockNum())
+				return
+			}
+		}
+	}
+	if cfg.View != self.config.View || cfg.N != self.config.N || cfg.C != self.config.C || cfg.BlockMsgDelay != self.config.BlockMsgDelay ||
```

Comment:
I think it's better to compare hash of serialised config, rather than compare all data member?